rskj-core/src/test/java/co/rsk/peg/BridgeSupportIT.java

```diff
@@ -155,73 +175,147 @@ void setUpOnEachTest() {
+        Files.copy(source, target, StandardCopyOption.REPLACE_EXISTING);
     }
 
     @Test
-    void testInitialChainHeadWithBtcCheckpoints() throws Exception {
-        BridgeConstants bridgeTestNetConstants = BridgeTestNetConstants.getInstance();
+    void initialChainHeadWithBtcCheckpoints_whenCheckpointsWith32BytesChainWork_before_RSKIP454_shouldFail() throws Exception {
+        // arrange
+        bridgeConstants = bridgeMainNetConstants;
+        activations = ActivationConfigsForTest.arrowhead631().forBlock(0);
+        arrangeCheckpointsSource("32-byte-chainwork.production.checkpoints");
 
-        Repository repository = createRepository();
-        Repository track = repository.startTracking();
+        // act
+        arrangeBridgeSupport();
 
-        BtcBlockStoreWithCache.Factory btcBlockStoreFactory = new RepositoryBtcBlockStoreWithCache.Factory(bridgeTestNetConstants.getBtcParams());
-        BridgeStorageProvider provider = new BridgeStorageProvider(track, PrecompiledContracts.BRIDGE_ADDR, bridgeTestNetConstants.getBtcParams(), activationsBeforeForks);
+        // assert
+        Assertions.assertThrows(IllegalArgumentException.class, bridgeSupport::getBtcBlockchainBestChainHeight, "The given number does not fit in 12");
+    }
+
+    @ParameterizedTest
+    @CsvSource(
+        {
+            "12-byte-chainwork.production.checkpoints",
+            "12-byte-chainwork.production.checkpoints",
+            "32-byte-chainwork.production.checkpoints",
+            "12-byte-chainwork-mix-format.production.checkpoints"
+        }
+    )
+    void initialChainHeadWithBtcCheckpoints_whenCheckpointsWith32BytesChainWork_after_RSKIP454_ok(
+        String checkpointFileName) throws Exception {
+        // arrange
+        bridgeConstants = bridgeMainNetConstants;
+        activations = ActivationConfigsForTest.lovell700().forBlock(0);
+        arrangeCheckpointsSource(checkpointFileName);
+        arrangeBridgeSupport();
+
+        // act
+        // Force instantiation of blockstore
+        bridgeSupport.getBtcBlockchainBestChainHeight();
+
+        // assert
+        assertGenesisFedCreationTimeIsCheckpointBestChainHeight();
+    }
+
+    private void assertGenesisFedCreationTimeIsCheckpointBestChainHeight()
+        throws BlockStoreException, IOException {
+        InputStream checkpointsStream = bridgeSupport.getCheckPoints();
+        CheckpointManager manager = new CheckpointManager(bridgeConstants.getBtcParams(),
+            checkpointsStream);
+        long genesisFedCreationTimeAdjusted = bridgeSupport.getActiveFederation().getCreationTime().toEpochMilli() - checkpointTimeAdjustmentForPossibleCLockDrift;
+        StoredBlock checkpoint = manager.getCheckpointBefore(genesisFedCreationTimeAdjusted);
+
+        assertEquals(checkpoint.getHeight(), bridgeSupport.getBtcBlockchainBestChainHeight());
+    }
+
+    private void arrangeBridgeSupport() {
```

Comment:
```suggestion
    private void arrangeBridgeSupport(bridgeConstants, activations) {
```

wdyt? to avoid doing the
```
bridgeConstants = specificConstants
activations = specificActivations
```

Reply:
btw, i'd set the bridgeSupport in the general setup, and create it again when needed in specific tests.
In general we've been trying to avoid using these type of methods

Reply:
This whole class needs a refactor for the other methods that are not being touched. But this is out of the scope of this task. 

Anyway, regarding using this approach for arrangement in a test, I think you may be referencing the name `arrangeBridgeSupport,` which could be different. I accept that. But this is the same approach we've been using lately. I got inspired by changes in the  `svp` project. E.g., https://github.com/rsksmart/rskj/blob/add-federation-context/rskj-core/src/test/java/co/rsk/peg/BridgeSupportSvpTest.java#L1148C18-L1148C49

Reply:
Added your suggestion